### ISSUE TITLE
Mocking https connections fails when pyOpenSSL is installed

### ISSUE THREAD

**Issue by kennydo** [mindflayer/python-mocket#65]
If I `pip install pyOpenSSL==17.5.0` and then run the `pytest tests/main/test_https.py`, I get an exception that looks like this:
```
if isinstance(e.reason, _SSLError):
                # This branch is for urllib3 v1.22 and later.
>               raise SSLError(e, request=request)
E               requests.exceptions.SSLError: HTTPSConnectionPool(host='testme.org', port=443): Max retries exceeded with url: /json (Caused by SSLError(SSLError("bad handshake: SysCallError(9, 'EBADF')",),))
```

urllib3 does some things differently when it detects that the OpenSSL module is available. I found prior art in getting mocking of pyOpenSSL in HTTPretty (https://github.com/gabrielfalcao/HTTPretty/pull/215/files) but it looks pretty gnarly.

Is adding support for environments with PyOpenSSL installed anywhere on the roadmap?

**Comment by mindflayer** [comment]
Actually I was not aware of the problem, so it was not on the road-map. Thank you for the notice.
Let me see if I can apply a similar approach.

**Comment by mindflayer** [comment]
Hi @kennydo, I was looking at the PR from `HTTPretty` and the problem with that approach is that `AttributeError: module 'socket' has no attribute '_fileobject'`.

**Comment by ohenrik** [comment]
@mindflayer have you looked anymore at a possible solution to this. It is impossible for me to use mocket at the moment since my project relies on pyopenssl through authobahn. So i cannot run my tests.

**Comment by mindflayer** [comment]
Hi @ohenrik, I spent some time trying to figure out how to do it but I did not find any feasible solution. I am also trying to fix #68, so feel free to take it, if you are willing to.

**Comment by mindflayer** [comment]
The issue has been solved, sorry for being able to work on it after months.
